### PR TITLE
476 international standing order submission

### PR DESCRIPTION
**Pull request by mariantiris**
Issue: https://github.com/SecureBankingAccessToolkit/SecureBankingAccessToolkit/issues/476

**New**: International Standing Order submission and details functional tests for versions 3.1.8, 3.1.9, 3.1.10

**General Bug fixes**:
- Fixed a general bug where the mandatory fields test for all kinds of APIs were not executed
- Fixed general issues on the InternationalScheduled payments tests 
    - wrong api defined
    - wrong operations defined
    - wrong link used in tests - urlWithInternationalPaymentId instead of urlWithInternationalScheduledPaymentId

### CODE REVIEW

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_8/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.forgerock.securebanking.framework.extensions.junit.CreateTppCallback
+import com.forgerock.securebanking.framework.extensions.junit.EnabledIfVersion
+import com.forgerock.securebanking.openbanking.uk.common.api.meta.obie.OBVersion
+import com.forgerock.uk.openbanking.tests.functional.payment.international.standing.orders.api.v3_1_8.GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetails
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest(val tppResource: CreateTppCallback.TppResource) {
+
```
Comment (jorgesanchezperez) [inline src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_8/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt:11]: a bit long name.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_9/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import com.forgerock.securebanking.framework.extensions.junit.CreateTppCallback
+import com.forgerock.securebanking.framework.extensions.junit.EnabledIfVersion
+import com.forgerock.securebanking.openbanking.uk.common.api.meta.obie.OBVersion
+import com.forgerock.uk.openbanking.tests.functional.payment.international.standing.orders.api.v3_1_8.GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetails
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest(val tppResource: CreateTppCallback.TppResource) {
+
```
Comment (jorgesanchezperez) [inline src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_9/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt:11]: A bit long name.

**Review by jorgesanchezperez — CHANGES_REQUESTED** [review]
Could we simplify some long classes names?

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_10/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.forgerock.securebanking.framework.extensions.junit.CreateTppCallback
+import com.forgerock.securebanking.framework.extensions.junit.EnabledIfVersion
+import com.forgerock.securebanking.openbanking.uk.common.api.meta.obie.OBVersion
+import com.forgerock.uk.openbanking.tests.functional.payment.international.standing.orders.api.v3_1_8.GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetails
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest(val tppResource: CreateTppCallback.TppResource) {
+
```
[the remark this answers is not in the document]
Comment (mariantiris) [reply src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_10/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt]: Renamed this class and others as well that were too long.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_8/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt (hunk 1 above)
Comment (mariantiris) [reply src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_8/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt]: Renamed this class and others as well that were too long.

File: src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_9/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt (hunk 2 above)
Comment (mariantiris) [reply src/test/kotlin/com/forgerock/uk/openbanking/tests/functional/payment/international/standing/orders/junit/v3_1_9/GetInternationalStandingOrderInternationalStandingOrderIdPaymentDetailsTest.kt]: Renamed this class and others as well that were too long.

**Comment by mariantiris** [comment]
@jorgesanchezperez I have renamed the classes that you reviewed and also similar classes that had the name too long to fit in the same naming convention for the payment details tests. The names should be much more clear now.

**Review by jorgesanchezperez — APPROVED** [review]

**Pull request merged**